src/main/scala/com/atomist/rug/runtime/js/JavaScriptHandlerFinder.scala

```diff
@@ -26,4 +36,51 @@ object JavaScriptHandlerFinder {
     jsc.engine.put("atomist", atomist)
     jsc.load(rugAs)
   }
+
+  def fromJavaScriptArchive(rugAs: ArtifactSource,
+                            ctx: JavaScriptHandlerContext,
+                            context: JavaScriptContext = null): Seq[SystemEventHandler] = {
+    val jsc: JavaScriptContext =
+      if (context == null)
+        new JavaScriptContext()
+      else
+        context
+
+    jsc.load(rugAs)
+    handlersFromVars(rugAs, jsc, ctx)
+  }
+
+  private def handlersFromVars(rugAs: ArtifactSource, jsc: JavaScriptContext, ctx: JavaScriptHandlerContext): Seq[SystemEventHandler] = {
+    jsc.vars.foldLeft(Seq[SystemEventHandler]())((acc: Seq[SystemEventHandler], jsVar) => {
+      val obj = jsVar.scriptObjectMirror
+      if(obj.hasMember("name") && obj.hasMember("description") && obj.hasMember("handle") && obj.hasMember("expression")){
+        val name = obj.getMember("name").asInstanceOf[String]
+        val description = obj.getMember("description").asInstanceOf[String]
+        val handle = obj.getMember("handle").asInstanceOf[ScriptObjectMirror]
+        val expression = obj.getMember("expression") match {
+          case x: String => x
+          case o: ScriptObjectMirror => o.getMember("expression").asInstanceOf[String]
+          case _ => Nil.asInstanceOf[String]
```

Comment:
Could the context parameter be `Option[JavaScriptContext]` rather than null? More idiomatic Scala